pallets/fiat_filter/rpc/src/lib.rs

```diff
@@ -0,0 +1,88 @@
+// use frame_support::Parameter;
+use jsonrpc_core::{Error as RpcError, ErrorCode, Result};
+use jsonrpc_derive::rpc;
+use sp_api::ProvideRuntimeApi;
+use sp_blockchain::HeaderBackend;
+use sp_core::Bytes;
+use sp_rpc::number::NumberOrHex;
+use sp_runtime::codec::{Codec, Decode};
+use sp_runtime::generic::BlockId;
+use sp_runtime::traits::{Block as BlockT, MaybeDisplay};
+use std::convert::TryInto;
+use std::sync::Arc;
+
+pub use fiat_filter_rpc_runtime_api::FiatFeeRuntimeApi;
+
+#[rpc]
+pub trait FiatFeeApi<Balance> {
+    /// Accepts a scale-encoded extrinsic, returns fee in µDOCK as Balance (u64)
+    #[rpc(name = "fiat_filter_getCallFeeDock")]
+    fn get_call_fee_dock(&self, encoded_xt: Bytes) -> Result<Balance>;
```

Comment:
RPC's allow you to query state at a specific block (current or past) so that you can know what the fee was at a certain time in past. All (ours and Substrate's that I have seen) current RPCs follow this convention and consumers of this RPC might expect the same. Any reason you wouldn't want to allow this?

Reply:
Is there a use case for checking what the fee was in the past ?
I didn't see one when writing the RPC, and thought the only use case would be to check the fee before making a call.
I might be wrong, I can add this parameter if needed.

If you have executed a call, you could see in the chain events how much was actually charged for it, right ?

Reply:
Not yet. I just wanted to be consistent with the other RPC calls.